WordPress/Classes/Models/Blog.m

```diff
@@ -292,9 +293,34 @@ - (NSString *)postFormatTextFromSlug:(NSString *)postFormatSlug
 // WP.COM private blog.
 - (BOOL)isPrivate
 {
-    return (self.isHostedAtWPcom && [[self getOptionValue:@"blog_public"] isEqualToString:@"-1"]);
+    return (self.isHostedAtWPcom && [self.privacy isEqualToNumber:@(BlogPrivacyPrivate)]);
 }
 
+- (NSString *)textForBlogPrivacy:(BlogPrivacy)privacy
+{
+    switch (privacy) {
+        case BlogPrivacyPrivate:
+            return NSLocalizedString(@"Private", @"Text for privacy settings: Private");
+            break;
+        case BlogPrivacyHidden:
+            return NSLocalizedString(@"Hidden", @"Text for privacy settings: Hidden");
+            break;
+        case BlogPrivacyPublic:
+            return NSLocalizedString(@"Public", @"Text for privacy settings: Public");
+            break;
+    }
+    return NSLocalizedString(@"Unknow", @"Text for unknow privacy setting");;
```

Comment:
Also I'm not sure this belongs in `Blog` - since it's a translation of BlogPrivacy and it's not specific to the instance of Blog. At a minimum I'd say get rid of the parameter and only let this return the text for the self.privacy value.